.github/workflows/workflow.yaml

```diff
@@ -0,0 +1,104 @@
+name: Workflow for install-docker
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - '*'
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  validate:
+    permissions:
+      contents: read
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    container: 
+      image: rancher/dapper:v0.6.0
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Validate
+        run: dapper validate
+
+  test:
+    permissions:
+      contents: read
+    needs: validate
+    runs-on: ubuntu-latest
+    timeout-minutes: 90
+    if: github.event_name == 'pull_request'
```

Comment:
I'm generally a fan of creating multiple workflow files for different actions. For example, one for PRs, another for merges and one for tagging. IMO that pattern is easier to read and maintain, gets rid of `skipped` checks in the github UI when raising PRs, and allows us to avoid the use of `if` statements depending on what action is invoking the workflow. Adding conditionals like this can make it harder to determine what should and shouldn't execute given a particular `on` statement. 

I don't work with this repo often, so I won't hold up this PR based off this, but it's something that should be considered.